### PR TITLE
Habit chart Backend to firebase

### PR DESCRIPTION
**Pull request by jakerogers2022**
## Trello Ticket :ticket: ##
https://trello.com/c/SEgJnqHC/42-be-habit-chart-backend-new-field-in-group-data
## Summary :chart_with_upwards_trend: ##
Changed Habit Chart to use data from firebase and work with missing dates
## Test plan :clipboard: ##
Working on Android

### CODE REVIEW

File: components/CommonCompHabitChart.js [hunk 1]
```diff
@@ -27,20 +28,61 @@ const gridRow = (row, clr) => {
...
-  //const recentHabit = [{'name': 'test0', 'recent': [1,0,1,0,0,0,1,1,0,0,1,1,0,0,1,1,1,0,1,1,1,0]}, {'name': 'test1', 'recent': [1,0,0,1,1,0,1,0,1,1,0,0,0,1,0,1,0,1,1,   1,0,0]}, {'name': 'test2', 'recent': [1,1,1,1,0,1,0,1,0,0,1,1,1,0,0,1,1,0,0,1,1,1]}];
+            startingDates.push(Math.min(...Object.keys(groupMembersData[key])));
+            dataList.push(groupMembersData[key]);
+        }
+    }
+    const moment = require('moment')
+
+    //This starts the habit chart at the earliest date any of the users completed the habit
+    let firstDay = ''+Math.min(...startingDates);
+    firstDay = (new Date(firstDay.substring(0,4), parseInt(firstDay.substring(4,6))-1, firstDay.substring(6)));
```
Comment (Peiyj) [inline components/CommonCompHabitChart.js:58]: nit: maybe add a comment `// attempt to get the earliest day in YYYYMMDD` for better clarity.

File: components/CommonCompHabitChart.js [hunk 2]
```diff
@@ -27,20 +28,61 @@ const gridRow = (row, clr) => {
...
+    let day = moment(firstDay).format('YYYY/MM/DD')
+    const today = moment().format('YYYY/MM/DD')
+
+    //datesSinceFirstDay is an array that holds all the dates since the earliest date recorded
+    const datesSinceFirstDay = [day.split('/').join('')];
+    let count = 0;
+    while(today !== day && count < 100){
```
Comment (Peiyj) [inline components/CommonCompHabitChart.js:65]: Is there a reason why the count is hardcoded to 100? Does that mean we are looking 100 days prior to today?

File: components/CommonCompHabitChart.js [hunk 3]
```diff
@@ -27,20 +28,61 @@ const gridRow = (row, clr) => {
...
+        day = moment(day).add(1, 'days').format('YYYY/MM/DD');
+        datesSinceFirstDay.push(day.split('/').join(''))
+        count = count + 1;
+    }
+    
+    //Outer loops through the users
+    //inner loops through all dates and checks if user completed habit on that date
+    const userDataList = []
```
Comment (Peiyj) [inline components/CommonCompHabitChart.js:73]: Nice :)!

File: screens/GroupInfo.js [hunk 4]
```diff
@@ -16,48 +16,54 @@ class GroupInfo extends Component {
...
+      var usersArray = [];
       const groupID = this.state.groupID;
-      firebase.database().ref('/groups/'+groupID).on('value', function (snapshot) {
-        snapshot.forEach(function (childSnapshot) {
-          groupArray.push(childSnapshot.toJSON());
-        });
+      this.setState({usernames: []})
+
+      firebase.database().ref('/').on('value', (snapshot) => {
+        const jsonSnap = snapshot.toJSON();
```
Comment (Peiyj) [inline screens/GroupInfo.js:44]: nit: I think rename `jsonSnap` to `firebaseDB` makes more sense?

File: screens/GroupInfo.js [hunk 5]
```diff
@@ -16,48 +16,54 @@ class GroupInfo extends Component {
...
       });
-      console.log('groupArray: ',groupArray);
-      this.setState({group: groupArray});
-      this.setState({streak: groupArray[6]});
+      
       
     }
     render() {
       const stack = createStackNavigator()
       //TODO get data from firebase
```
Comment (Peiyj) [inline screens/GroupInfo.js:59]: nit: can remove this todo now :)

**Review by Peiyj — APPROVED** [review]
LGTM! Just a few minor suggestions :) Overall, amazing work Jake!

File: components/CommonCompHabitChart.js (hunk 2 above)
Comment (jakerogers2022) [reply components/CommonCompHabitChart.js:65]: Yeah this is just so the habit chart won't show more than 100 days

**Pull request merged**